### DISCUSSION TITLE
Install created lib to specific folder in build/ doesn't seem to work?

### DISCUSSION THREAD

**Discussion by avetiso** [mesonbuild/meson#10088] [Q&A] [answered]
```
libbase64 = static_library('base64',
                            sources: ['_internal/base64.cpp'],
                            include_directories: inc,
                            install: true,
                            install_dir: meson.global_build_root() / 'lib')
```

I'm trying to make sure this library ends up in build/lib but no matter what I try, it always ends in src/ (the meson.build file this comes from also src/). I've tried all sorts of variation on this logic and have also tried passing the libdir as an arg to meson.configure() in Conan, with no luck. I've also tried meson.current_source_dir() as well as source_root().

I want the final file structure after the packaging is complete to look like this:

```
src/
     base64/
          base64.h
lib/
     libbase64.a
```

lib should not exist in the source tree, it should be created when the lib is created only. 

No matter what I try, the build directory ends up like this:

```
src/
     libbase64.a
     libebase64.a.p/
```

which means my packaging step doesn't find the lib because it looks for lib/ to copy it out of.

I want the library to end up `lib/libbase64.a` not `src/libbase64.a`.

How can I get the lib into a folder dynamically at build time into a folder called lib/ in the build directory? install_dir doesn't seem to work, or at the very least is feels very unintuitive.

**Comment by avetiso** [comment]
Here is another attempt with the some logging:

```
install_dir = join_paths(meson.global_build_root(), 'lib')

warning('HERE')
warning(install_dir)

libbase64 = static_library('base64',
                            sources: ['_internal/base64.cpp'],
                            include_directories: inc,
                            install: true,
                            install_dir: install_dir)
```

Logs print:
```
src/meson.build:5: WARNING: HERE
src/meson.build:6: WARNING: /Users/dario.avetisov/dev/conan/base64/build/lib
Build targets in project: 2
```

No lib/ folder is created, and the library still gets placed into src/.

**Comment by rusty-snake** [comment, marked as answer]
Did you run `meson install` to install your library? Or better why do you want to install it into your builddir?

**Reply by avetiso** [reply]
That was it, thank you. The reason we I want to install it into lib/* is because all 50+ packages we host have a generic packaging tool that looks for a folder called lib/, and I want to set things up to follow already established tooling.
